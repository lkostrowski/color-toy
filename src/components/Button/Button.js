import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import colors from "../../colors";

const ButtonWrapper = glamorous.button({
    background: colors.green,
    width: '100%',
    display: 'block',
    color: colors.light,
    height: '3rem',
    textTransform: 'uppercase',
    fontSize: '1.6rem',
    cursor: 'pointer',
    ':focus': {
        boxShadow: 'none',
        outline: 'none'
    }
});

const Button = ({children, ...props}) => (
    <ButtonWrapper {...props}>
        {children}
    </ButtonWrapper>
);

Button.propTypes = {
    children: PropTypes.any
};

export default Button;
