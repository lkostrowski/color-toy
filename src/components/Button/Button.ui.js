import glamorous from "glamorous";
import colors from "../../colors";

export const ButtonWrapper = glamorous.button({
    background: colors.green,
    width: '100%',
    display: 'block',
    color: colors.light,
    height: '3rem',
    textTransform: 'uppercase',
    fontSize: '1.6rem',
    cursor: 'pointer',
    ':focus': {
        boxShadow: 'none',
        outline: 'none'
    }
});
