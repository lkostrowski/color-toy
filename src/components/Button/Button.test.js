import Button from './Button';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Button component', () => {
    it('render expected html', () => {
        const markup = renderer.create(
            <Button>Test</Button>
        ).toJSON();

        expect(markup).toMatchSnapshot();
    });
});
