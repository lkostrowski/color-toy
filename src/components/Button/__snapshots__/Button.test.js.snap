// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render expected html 1`] = `
<button
  className="css-1kigtsu"
>
  Test
</button>
`;
